lightning/src/ln/onion_utils.rs

```diff
@@ -615,8 +639,9 @@ pub(super) fn process_onion_failure<T: secp256k1::Signing, L: Deref>(
 							} else {
 								// The node in question intentionally encoded a 0-length channel update. This is
 								// likely due to https://github.com/ElementsProject/lightning/issues/6200.
+								short_channel_id = Some(failing_route_hop.short_channel_id);
 								network_update = Some(NetworkUpdate::ChannelFailure {
-									short_channel_id: route_hop.short_channel_id,
+									short_channel_id: failing_route_hop.short_channel_id,
```

Comment:
We only mark the channel as disabled if `is_permanent` is set. It's nice to set it because then we can keep the behavior of defaulting to `NetworkUpdate::NodeFailure` on a totally bogus channel update.